--- /tmp/orig-6638648519863966173.java	2025-09-24 22:29:54.749402781 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/ml/liblinear/Liblinear.java	2025-09-24 22:29:51.061144956 +0000
@@ -654,7 +654,19 @@
 	}
 	
 	protected InputStream getInputStreamFromConfigFileEntry(String suffix) throws MaltChainedException {
-		return getConfiguration().getConfigurationDir().getInputStreamFromConfigFileEntry(owner.getModelName()+getLearningMethodName()+suffix);
+	org.maltparser.core.config.ConfigurationDir __arodnap_temp0 = null;
+	try {
+    __arodnap_temp0 = getConfiguration().getConfigurationDir();
+    return __arodnap_temp0.getInputStreamFromConfigFileEntry(owner.getModelName()+getLearningMethodName()+suffix);
+} finally {
+    if (__arodnap_temp0 != null) {
+        try {
+            __arodnap_temp0.terminate();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
 	}
 	
 	
