--- /tmp/orig-13071195082857695537.java	2025-09-24 21:55:03.631225476 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url47af6b1535_rolandotr_db_comparison_tgz-pJ8-methodology_HistoryJ8/src/protocols/specifications/TMAProtocol.java	2025-09-24 21:55:01.939107190 +0000
@@ -244,14 +244,15 @@
 		int r;
 		distances2 = new double[n];
 		File f = new File(getDistanceFileNameOfValues());
-		BufferedReader reader = new BufferedReader(new FileReader(f));
-		while (reader.ready()){
+		try (java.io.FileReader __arodnap_temp0 = new FileReader(f)) {
+		    BufferedReader reader = new BufferedReader(__arodnap_temp0);
+		    while (reader.ready()){
 			tmp = reader.readLine();
 			split = tmp.split(" ");
 			if (Integer.parseInt(split[0]) == n) break;
 			distances2[Integer.parseInt(split[0])] = Double.parseDouble(split[1]);
 		}
-		reader.close();
+		}
 	}	
 
 	
