--- /tmp/orig-4752031602985170944.java	2025-09-24 22:17:09.203888085 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url7f15a160d2_timofei7_old_ai_assignments_tgz-pJ8-org_json_TestJ8/src/hw5/P3bMain.java	2025-09-24 22:17:06.975732329 +0000
@@ -193,9 +193,9 @@
 		}
 			
 		//TODO: do I need to be writing this out to file?
-		try
-		{
-			BufferedWriter out = new BufferedWriter(new FileWriter(filename));
+		try(
+java.io.FileWriter __arodnap_temp0 = new FileWriter(filename)) 		{
+			BufferedWriter out = new BufferedWriter(__arodnap_temp0);
 			out.write(forfile);
 			out.close();
 		}
