--- /tmp/orig-2554274374344693516.java	2025-09-24 22:41:32.666190739 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/dataaccess/SerializeDB.java	2025-09-24 22:41:31.098081115 +0000
@@ -59,8 +59,7 @@
 	}
 	
 	public void saveData() {
-		try {
-			FileOutputStream fout = new FileOutputStream(fileName);
+		try (FileOutputStream fout = new FileOutputStream(fileName)) {
 			ObjectOutputStream oos = new ObjectOutputStream(fout);
 			
 			oos.writeObject(cineplexes);
@@ -311,4 +310,4 @@
 		return staffID;
 	}
 	
-}
+}
\ No newline at end of file
