--- /tmp/orig-15782018196051538946.java	2025-09-24 22:21:57.876067266 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/core/flow/system/FlowChartSystem.java	2025-09-24 22:21:54.135805808 +0000
@@ -57,12 +57,12 @@
 	}
 	
 	public void load(URL specModelURL) throws MaltChainedException {
-        try {
+        try (java.io.InputStream __arodnap_temp0 = specModelURL.openStream()) {
             DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
             DocumentBuilder db = dbf.newDocumentBuilder();
             Element root = null;
 
-            root = db.parse(specModelURL.openStream()).getDocumentElement();
+            root = db.parse(__arodnap_temp0).getDocumentElement();
 
             if (root == null) {
             	throw new FlowException("The flow chart system file '"+specModelURL.getFile()+"' cannot be found. ");
