--- /tmp/orig-15002071167137565810.java	2025-09-24 21:33:57.154712084 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url0307c0e995_js_labs_js_collider_tgz-pJ8-org_jsl_tests_shmem_throughput_MainJ8/src/org/jsl/tests/echo_throughput/Client.java	2025-09-24 21:33:54.922556110 +0000
@@ -42,9 +42,8 @@
     {
         public void run()
         {
-            try
-            {
-                final SocketChannel socketChannel = SocketChannel.open( m_addr );
+            try(
+final SocketChannel socketChannel = SocketChannel.open( m_addr ))             {
                 final Socket socket = socketChannel.socket();
                 socket.setTcpNoDelay( true );
                 socket.setSendBufferSize( m_socketBufferSize );
