--- /tmp/orig-12214793479573205185.java	2025-09-24 22:53:18.299520544 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urldae1c57e59_mustafaakin_izuna_mapeditor_tgz-pJ8-in_mustafaak_izuna_actions_ActionTestJ8/src/in/mustafaak/izuna/actions/ActionTest.java	2025-09-24 22:53:17.187442800 +0000
@@ -62,8 +62,7 @@
 	}
 
 	public static void streamWave(WaveInfo waveInfo) {
-		try {
-			Socket socket = new Socket();
+		try (Socket socket = new Socket()) {
 			socket.connect(new InetSocketAddress("192.168.1.104", 5000), 50000);
 			Persister serializer = new Persister();
 			OutputStream os = socket.getOutputStream();
