--- /tmp/orig-12202974943211300694.java	2025-09-24 22:36:08.311516400 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url888b6fd9b4_jmacglashan_commandsToTasks_tgz-pJ8-commands_tests_Model3TestsJ8/src/commands/auxiliarytools/TrajectoryRecorder.java	2025-09-24 22:36:06.007355329 +0000
@@ -457,8 +457,8 @@
 		String teSRep = parser.getStringOfTrainingElement(te);
 		
 		//write this string out
-		try{
-			BufferedWriter out = new BufferedWriter(new FileWriter(path));
+		try(java.io.FileWriter __arodnap_temp0 = new FileWriter(path)) {
+			BufferedWriter out = new BufferedWriter(__arodnap_temp0);
 			out.write(teSRep);
 			out.close();
 		}catch(Exception E){
