--- /tmp/orig-12830895825197924948.java	2025-09-24 22:55:34.361032406 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/yuccasrv/Yucca.java	2025-09-24 22:55:31.808853988 +0000
@@ -278,6 +278,14 @@
 	    
 	    System.err.println( "[IOException] Failed to print warranty: " + e.getMessage() );
 	    
+	} finally {
+	    if (reader != null) {
+	        try {
+	            reader.close();
+	        } catch (Exception e) {
+	            e.printStackTrace();
+	        }
+	    }
 	} 
     }
 
