--- /tmp/orig-7030773084954559580.java	2025-09-24 21:58:36.230087054 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/io/image/ImagePersistence.java	2025-09-24 21:58:34.013932135 +0000
@@ -566,9 +566,8 @@
     */
     public static boolean exportPNM(File fd, Image img)
     {
-        try {
+        try (FileOutputStream fos = new FileOutputStream(fd)) {
             BufferedOutputStream writer;
-            FileOutputStream fos = new FileOutputStream(fd);
             writer = new BufferedOutputStream(fos);
 
             String line1 = "P5\n";
