--- /tmp/orig-6192216134036399181.java	2025-09-24 22:15:25.416632998 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/io/IoServerTest.java	2025-09-24 22:15:23.480497656 +0000
@@ -292,11 +292,12 @@
     public void testAddConnectionListener()
     {
         ServerConnectionListener listener = null;
-        IoServer instance = new IoServer();
-        instance.addConnectionListener(listener);
-        assertEquals(0, instance.connectionListeners.size());
-
-        listener = new ServerConnectionListener()
+        ping.pong.net.server.io.IoServer instance = null;
+        try {
+            instance = new IoServer();
+            instance.addConnectionListener(listener);
+            assertEquals(0, instance.connectionListeners.size());
+            listener = new ServerConnectionListener()
         {
             @Override
             public void connectionAdded(Server server, Connection conn)
@@ -312,9 +313,19 @@
                 assertNotNull(conn);
             }
         };
+            instance.addConnectionListener(listener);
+            assertEquals(1, instance.connectionListeners.size());
+        } finally {
+            if (instance != null) {
+                try {
+                    instance.shutdown();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
 
-        instance.addConnectionListener(listener);
-        assertEquals(1, instance.connectionListeners.size());
     }
 
     /**
