--- /tmp/orig-15706468163777542458.java	2025-09-24 22:21:13.236946838 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/core/options/OptionManager.java	2025-09-24 22:21:09.584691532 +0000
@@ -261,12 +261,22 @@
 	 * @param fileName	The path to the file where the saveOption should by saved.
 	 */
 	public void saveOptions(String fileName) throws MaltChainedException { 
-		try {
-			saveOptions(new OutputStreamWriter(new FileOutputStream(fileName), "UTF-8"));
+	java.io.FileOutputStream __arodnap_temp0 = null;
+	try {
+	__arodnap_temp0 = new FileOutputStream(fileName);
+		saveOptions(new OutputStreamWriter(__arodnap_temp0, "UTF-8"));
 		} catch (FileNotFoundException e) {
 			throw new OptionException("The file '"+fileName+"' cannot be created. ", e);
 		} catch (UnsupportedEncodingException e) {
 			throw new OptionException("The charset 'UTF-8' is unsupported. ", e);
+		} finally {
+		    if (__arodnap_temp0 != null) {
+		        try {
+		            __arodnap_temp0.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
 		}
 		
 	}
