--- /tmp/orig-12579965368317696383.java	2025-09-24 22:16:19.748430978 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/connection/io/AbstractIoConnectionTest.java	2025-09-24 22:16:17.856298712 +0000
@@ -61,12 +61,42 @@
     {
         AbstractIoConnection instance = new AbstractIoConnectionImpl();
         assertFalse(instance.isUsingCustomSerialization());
-        instance = new AbstractIoConnectionImpl(null, new ReadFullyDataReader(), null, null);
-        assertTrue(instance.isUsingCustomSerialization());
-        instance = new AbstractIoConnectionImpl(null, new ReadFullyDataReader(), new WriteByteArrayDataWriter(), null);
-        assertTrue(instance.isUsingCustomSerialization());
-        instance = new AbstractIoConnectionImpl(null, null, new WriteByteArrayDataWriter(), null);
-        assertTrue(instance.isUsingCustomSerialization());
+        try {
+            instance = new AbstractIoConnectionImpl(null, new ReadFullyDataReader(), null, null);
+            assertTrue(instance.isUsingCustomSerialization());
+        } finally {
+            if (instance != null) {
+                try {
+                    instance.close();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        try {
+            instance = new AbstractIoConnectionImpl(null, new ReadFullyDataReader(), new WriteByteArrayDataWriter(), null);
+            assertTrue(instance.isUsingCustomSerialization());
+        } finally {
+            if (instance != null) {
+                try {
+                    instance.close();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+        try {
+            instance = new AbstractIoConnectionImpl(null, null, new WriteByteArrayDataWriter(), null);
+            assertTrue(instance.isUsingCustomSerialization());
+        } finally {
+            if (instance != null) {
+                try {
+                    instance.close();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
     }
 
     /**
