--- /tmp/orig-4024897365487667034.java	2025-09-24 22:11:42.485049314 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ThreadStarter.java	2025-09-24 22:11:40.752928233 +0000
@@ -280,11 +280,13 @@
 		out.mkdirs();
 		if (combs == null || combs.size() < 1)
 			return;
+		java.io.FileOutputStream __arodnap_temp1 = null;
+		java.io.FileOutputStream __arodnap_temp0 = null;
 		try {
 			long time = System.currentTimeMillis();
+			__arodnap_temp0 = new FileOutputStream("_out" + File.separator + time + ".csv");
 			BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(
-					new FileOutputStream("_out" + File.separator + time
-							+ ".csv"), "UTF-8"));
+					__arodnap_temp0, "UTF-8"));
 			bw.write(Combo.OUTPUT_HEAD);
 			bw.newLine();
 			for (int i = 0; i < combs.size(); i++) {
@@ -293,17 +295,32 @@
 				bw.newLine();
 			}
 			bw.close();
+			__arodnap_temp1 = new FileOutputStream("_out" + File.separator + time + ".txt");
 
 			if (!noinfo) {
 				BufferedWriter bufferedWriter = new BufferedWriter(
-						new OutputStreamWriter(new FileOutputStream("_out"
-								+ File.separator + time + ".txt"), "UTF-8"));
+						new OutputStreamWriter(__arodnap_temp1, "UTF-8"));
 				bufferedWriter.write(infos);
 				bufferedWriter.close();
 			}
 
 		} catch (Exception e) {
 			e.printStackTrace();
+		} finally {
+		    if (__arodnap_temp1 != null) {
+		        try {
+		            __arodnap_temp1.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
+		    if (__arodnap_temp0 != null) {
+		        try {
+		            __arodnap_temp0.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
 		}
 
 	}
