--- /tmp/orig-3030385955516060308.java	2025-09-24 21:53:57.434598047 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url47af6b1535_rolandotr_db_comparison_tgz-pJ8-methodology_HistoryJ8/src/utils/Report.java	2025-09-24 21:53:55.634472210 +0000
@@ -198,11 +198,12 @@
 
 	private static List<DBProtocol> getSKIProtocol(File f) throws NumberFormatException, IOException {
 		List<DBProtocol> protocols = new LinkedList<>();
-		BufferedReader reader = new BufferedReader(new FileReader(f));
-		String regex = "SKI-\\{(\\d+), (\\d+)\\}";
-		Pattern pattern = Pattern.compile(regex);
-		String retrievedText = null;
-		for (String line = reader.readLine(); line  != null; line = reader.readLine()){
+		try (java.io.FileReader __arodnap_temp0 = new FileReader(f)) {
+		    BufferedReader reader = new BufferedReader(__arodnap_temp0);
+		    String regex = "SKI-\\{(\\d+), (\\d+)\\}";
+		    Pattern pattern = Pattern.compile(regex);
+		    String retrievedText = null;
+		    for (String line = reader.readLine(); line  != null; line = reader.readLine()){
 			Matcher m = pattern.matcher(line);
 			while (m.find()) {
 				SKIProtocol p = new SKIProtocol(Integer.parseInt(m.group(2)), DBProtocol.SIZE_OF_NONCES);
@@ -210,7 +211,7 @@
 				protocols.add(p);
 			}
 		}
-		reader.close();
+		}
 		return protocols;
 	}
 
