--- /tmp/orig-5888421390938591964.java	2025-09-24 22:23:17.305619730 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/core/symbol/trie/TrieSymbolTableHandler.java	2025-09-24 22:23:13.597360507 +0000
@@ -103,12 +103,22 @@
 	}
 	
 	public void save(String fileName, String charSet) throws MaltChainedException  {
-		try {
-			save(new OutputStreamWriter(new FileOutputStream(fileName), charSet));
+	java.io.FileOutputStream __arodnap_temp0 = null;
+	try {
+	__arodnap_temp0 = new FileOutputStream(fileName);
+		save(new OutputStreamWriter(__arodnap_temp0, charSet));
 		} catch (FileNotFoundException e) {
 			throw new SymbolException("The symbol table file '"+fileName+"' cannot be created. ", e);
 		} catch (UnsupportedEncodingException e) {
 			throw new SymbolException("The char set '"+charSet+"' is not supported. ", e);
+		} finally {
+		    if (__arodnap_temp0 != null) {
+		        try {
+		            __arodnap_temp0.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
 		}
 	}
 	
