--- /tmp/orig-12929754589814884534.java	2025-09-24 22:38:26.637185968 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/c64utils/Assembler.java	2025-09-24 22:38:24.477034964 +0000
@@ -341,9 +341,8 @@
   }
 
   private void loadBinary(String binary) {
-    try {
+    try (InputStream fs = openBinary(workingDir + binary)) {
       if (workingDir == null) workingDir = "";
-      InputStream fs = openBinary(workingDir + binary);
       if (fs == null) return;
       int data = 0;
       int initPos = pos;
