--- /tmp/orig-4269340231017426448.java	2025-09-24 21:35:54.018878043 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url0d38a10ee7_diegopliebana_GEBT_tgz-pJ8-ch_idsia_scenarios_test_EvolveWithChangingSeedsJ8/src/wox/serial/Easy.java	2025-09-24 21:35:51.094673712 +0000
@@ -18,11 +18,10 @@
  */
 public class Easy {
     public static void save(Object ob, String filename) {
-        try {
+        try (FileWriter file = new FileWriter(filename)) {
             ObjectWriter writer = new SimpleWriter();
             Element el = writer.write(ob);
             XMLOutputter out = new XMLOutputter(); // ("  ", true);
-            FileWriter file = new FileWriter(filename);
             out.output(el, file);
             file.close();
             System.out.println("Saved object to " + filename);
