--- /tmp/orig-7381457672042041922.java	2025-09-24 22:43:57.344304976 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/factories/JsonGridFactoryTest.java	2025-09-24 22:43:55.028143056 +0000
@@ -83,12 +83,24 @@
 	public void testNoDetector() {
 		Properties properties = Configuration.createDefaults();
 		properties.setProperty(Configuration.PROP_USE_FAILURE_DETECTOR, Boolean.FALSE.toString());
-		
-		JsonGridFactory factory = new JsonGridFactory(
+		java.io.InputStream __arodnap_temp0 = null;
+		try {
+    __arodnap_temp0 = getClass().getResourceAsStream("grid-example.conf");
+    JsonGridFactory factory = new JsonGridFactory(
 				properties, 
-				getClass().getResourceAsStream("grid-example.conf"));
+				__arodnap_temp0);
+    factory.createGrid();
+} finally {
+    if (__arodnap_temp0 != null) {
+        try {
+            __arodnap_temp0.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
+
 		
-		factory.createGrid();
 	}
 	
 	@Test
