--- /tmp/orig-3136415064545544570.java	2025-09-24 21:40:30.130171508 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url11fb0973cf_raichuman9_Swimmer526_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/ubc/swim/tests/SliderCrankTest.java	2025-09-24 21:40:27.357977799 +0000
@@ -154,8 +154,9 @@
 		
 		addTextLine("Keys: (f) toggle friction, (m) toggle motor");
 		float torque = m_joint1.getMotorTorque();
-		Formatter f = new Formatter();
-		addTextLine(f.format("Friction: %b, Motor Force = %5.0f, ", m_joint2.isMotorEnabled(), torque).toString());
+		try (Formatter f = new Formatter()) {
+		    addTextLine(f.format("Friction: %b, Motor Force = %5.0f, ", m_joint2.isMotorEnabled(), torque).toString());
+		}
 
 	}
 	
