--- /tmp/orig-6064277979643956645.java	2025-09-24 22:13:51.606075318 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url7a45f192ca_mfullen_ping_pong_net_tgz-pJ8-ping_pong_net_client_io_TestClientNonPPNServerJ8/src/ping/pong/net/server/ClientExample.java	2025-09-24 22:13:49.865953677 +0000
@@ -196,17 +196,18 @@
                                                            CertificateException,
                                                            URISyntaxException
     {
-        Socket client = new Socket("localhost", 5011);
-        ObjectInputStream inputStream = new ObjectInputStream(client.getInputStream());
-        int readInt = inputStream.readInt();
-        System.out.println("I was assigned Client id: " + readInt);
-        ObjectOutputStream outputStream = new ObjectOutputStream(client.getOutputStream());
-        outputStream.writeObject(new byte[]
+        try (Socket client = new Socket("localhost", 5011)) {
+            ObjectInputStream inputStream = new ObjectInputStream(client.getInputStream());
+            int readInt = inputStream.readInt();
+            System.out.println("I was assigned Client id: " + readInt);
+            ObjectOutputStream outputStream = new ObjectOutputStream(client.getOutputStream());
+            outputStream.writeObject(new byte[]
                 {
                     1, 3, 2, 3
                 });
-        outputStream.flush();
-        outputStream.close();
+            outputStream.flush();
+            outputStream.close();
+        }
     }
 
     public static void regularConnectionByteArrayStaysOpen() throws IOException,
