--- /tmp/orig-7675091516288322663.java	2025-09-24 21:41:40.335077125 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url147ce07ebe_JoshuaKissoon_Kademlia_tgz-pJ8-kademlia_simulations_RoutingTableStateTestingJ8/src/kademlia/simulations/ContentSendingTest.java	2025-09-24 21:41:38.734965316 +0000
@@ -19,10 +19,11 @@
 
     public static void main(String[] args)
     {
-        try
+    kademlia.JKademliaNode kad1 = null;
+    try
         {
-            /* Setting up 2 Kad networks */
-            JKademliaNode kad1 = new JKademliaNode("JoshuaK", new KademliaId("ASF45678947584567467"), 7574);
+            
+            kad1 = new JKademliaNode("JoshuaK", new KademliaId("ASF45678947584567467"), 7574);
             System.out.println("Created Node Kad 1: " + kad1.getNode().getNodeId());
             JKademliaNode kad2 = new JKademliaNode("Crystal", new KademliaId("ASERTKJDHGVHERJHGFLK"), 7572);
             System.out.println("Created Node Kad 2: " + kad2.getNode().getNodeId());
@@ -55,6 +56,14 @@
         catch (IOException | ContentNotFoundException e)
         {
             e.printStackTrace();
+        } finally {
+            if (kad1 != null) {
+                try {
+                    kad1.shutdown(false);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
         }
     }
 }
