--- /tmp/orig-7861741470747534336.java	2025-09-24 22:44:42.507462268 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/Main.java	2025-09-24 22:44:40.235303424 +0000
@@ -21,15 +21,27 @@
 		properties.put(Configuration.PROP_BROKER_MAX_REPLICAS, "1");
 		properties.put(Configuration.PROP_USE_FAILURE_DETECTOR, Boolean.FALSE.toString());
 		properties.put(Configuration.PROP_BROKER_SCHEDULER_INTERVAL, "10");
-		
-		OurSim ourSim = new OurSim(
-				new FileEventProxy(new FileInputStream("resources/event-example.conf")), 
+		java.io.FileInputStream __arodnap_temp0 = null;
+		try {
+		    __arodnap_temp0 = new FileInputStream("resources/event-example.conf");
+		    OurSim ourSim = new OurSim(
+				new FileEventProxy(__arodnap_temp0), 
 				new JsonGridFactory(properties, new FileInputStream("resources/grid-example.conf")).createGrid(),
 				properties,
 				new BlankNetwork(),
 				new DefaultTraceCollector(new FileOutputStream("trace.out")));
+		    ourSim.run();
+		} finally {
+		    if (__arodnap_temp0 != null) {
+		        try {
+		            __arodnap_temp0.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
+		}
+
 		
-		ourSim.run();
 		
 		long end = System.currentTimeMillis();
 		
