--- /tmp/orig-17275936102887959216.java	2025-09-24 22:02:56.572286098 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url53f943c5cb_Praznat_AGP_tgz-pJ8-AMath_TestingJ8/src/Game/DefFile.java	2025-09-24 22:02:53.132045610 +0000
@@ -16,11 +16,11 @@
     public static String[] readFile (String fn) {
 	    File file = new File( fn ); 
 	    if ( file.exists() )  {
-            try
-			{
+            try(
+java.io.FileReader __arodnap_temp0 = new FileReader(file)) 			{
 			    String[] data = new String[maxfilelength];
 		        BufferedReader inFile = new BufferedReader(
-		            new FileReader( file ) );	
+		            __arodnap_temp0 );	
 		        int lineNum = 0;
 		        String line = inFile.readLine();
 		        while ( line != null )
