--- /tmp/orig-5581258297954250704.java	2025-09-24 22:27:55.513067641 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/ml/libsvm/Libsvm.java	2025-09-24 22:27:51.744804224 +0000
@@ -582,7 +582,19 @@
 	}
 	
 	protected InputStreamReader getInstanceInputStreamReader(String suffix) throws MaltChainedException {
-		return getConfiguration().getConfigurationDir().getInputStreamReader(owner.getModelName()+getLearningMethodName()+suffix);
+	org.maltparser.core.config.ConfigurationDir __arodnap_temp0 = null;
+	try {
+    __arodnap_temp0 = getConfiguration().getConfigurationDir();
+    return __arodnap_temp0.getInputStreamReader(owner.getModelName()+getLearningMethodName()+suffix);
+} finally {
+    if (__arodnap_temp0 != null) {
+        try {
+            __arodnap_temp0.terminate();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
 	}
 	
 	protected InputStreamReader getInstanceInputStreamReaderFromConfigFile(String suffix) throws MaltChainedException {
