--- /tmp/orig-12047063900815013503.java	2025-09-24 21:33:29.548783100 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url0307c0e995_js_labs_js_collider_tgz-pJ8-org_jsl_tests_shmem_throughput_MainJ8/src/org/jsl/tests/send_throughput/Client.java	2025-09-24 21:33:27.332628246 +0000
@@ -40,9 +40,8 @@
     {
         public void run()
         {
-            try
-            {
-                SocketChannel socketChannel = SocketChannel.open();
+            try(
+SocketChannel socketChannel = SocketChannel.open())             {
                 socketChannel.socket().setReceiveBufferSize( m_socketBufferSize );
 
                 StreamDefragger streamDefragger = new StreamDefragger(4)
