--- /tmp/orig-1691350370308544586.java	2025-09-24 21:53:29.716660442 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url47087446e9_kellrott_SparqlScape_tgz-pJ8-org_json_TestJ8/src/org/sparqlscape/SearchTest.java	2025-09-24 21:53:27.928535444 +0000
@@ -46,15 +46,12 @@
 */
 	
 	public static void main(String args[]) throws IOException {	
-		
-		CSVParser p = new CSVParser( new InputStreamReader( new FileInputStream(new File(args[0])) ), new CSVStrategy('\t', '"',	'#'));
-		
-		String[] head = p.getLine();
-
-		List< Map<String,Object> > editList = new LinkedList<Map<String,Object>>();
-
-		String []curRead = null;
-		do {
+	    try (java.io.FileInputStream __arodnap_temp0 = new FileInputStream(new File(args[0]))) {
+	        CSVParser p = new CSVParser( new InputStreamReader( __arodnap_temp0 ), new CSVStrategy('\t', '"',	'#'));
+	        String[] head = p.getLine();
+	        List< Map<String,Object> > editList = new LinkedList<Map<String,Object>>();
+	        String []curRead = null;
+	        do {
 			curRead = p.getLine();
 			System.out.println( curRead );
 			if ( curRead != null ) {
@@ -65,12 +62,17 @@
 				editList.add(features);
 			}			
 		} while ( curRead != null );
-		
-		SparqlInterface iSparql = new SparqlInterface( "http://proteins:8890/sparql" );		
-		for ( LinkTriple link :	iSparql.NodeSearch(null, editList, head[0]) ) {
+	        SparqlInterface iSparql = new SparqlInterface( "http://proteins:8890/sparql" );
+	        for ( LinkTriple link :	iSparql.NodeSearch(null, editList, head[0]) ) {
 			System.out.println( link );
 		}
-		//System.exit(0);		
+	    }
+	
+
+
+
+
+//System.exit(0);		
 	}
 	
 }
