--- /tmp/orig-12524299875637797228.java	2025-09-24 21:41:07.704797061 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url147ce07ebe_JoshuaKissoon_Kademlia_tgz-pJ8-kademlia_simulations_RoutingTableStateTestingJ8/src/kademlia/simulations/ContentUpdatingTest.java	2025-09-24 21:41:06.176690283 +0000
@@ -18,10 +18,11 @@
 
     public static void main(String[] args)
     {
-        try
+    kademlia.JKademliaNode kad1 = null;
+    try
         {
-            /* Setting up 2 Kad networks */
-            JKademliaNode kad1 = new JKademliaNode("JoshuaK", new KademliaId("ASF45678947584567467"), 7574);
+            
+            kad1 = new JKademliaNode("JoshuaK", new KademliaId("ASF45678947584567467"), 7574);
             System.out.println("Created Node Kad 1: " + kad1.getNode().getNodeId());
             JKademliaNode kad2 = new JKademliaNode("Crystal", new KademliaId("ASERTKJDHGVHERJHGFLK"), 7572);
             System.out.println("Created Node Kad 2: " + kad2.getNode().getNodeId());
@@ -54,6 +55,14 @@
         catch (IOException | ContentNotFoundException e)
         {
             e.printStackTrace();
+        } finally {
+            if (kad1 != null) {
+                try {
+                    kad1.shutdown(false);
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
         }
     }
 }
