--- /tmp/orig-5401245950030295495.java	2025-09-24 22:00:54.135727254 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/Compare/buildIndexDP.java	2025-09-24 22:00:51.379534584 +0000
@@ -74,16 +74,15 @@
 	public static void buildIndex(String sourceFileName, String destFileName, int fromId, int toId) throws IOException{
 		loadData(sourceFileName);
 		if(toId == -1)toId = points.size();
-		
-		final DataOutputStream dos_idx =  new DataOutputStream(new BufferedOutputStream(new FileOutputStream(new File(destFileName + ".idx"))));
-		final DataOutputStream dos_dat = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(new File(destFileName + ".dat"))));
-		
-		final int[] lock = new int[1];
-		final long[] w_lock = new long[1];
-		final int limit = toId; 
-		lock[0] = fromId;
-		w_lock[0] = 0;
-		for(int id = 0; id < ThreadNum; id ++){
+		try (java.io.FileOutputStream __arodnap_temp0 = new FileOutputStream(new File(destFileName + ".idx"))) {
+		    final DataOutputStream dos_idx =  new DataOutputStream(new BufferedOutputStream(__arodnap_temp0));
+		    final DataOutputStream dos_dat = new DataOutputStream(new BufferedOutputStream(new FileOutputStream(new File(destFileName + ".dat"))));
+		    final int[] lock = new int[1];
+		    final long[] w_lock = new long[1];
+		    final int limit = toId;
+		    lock[0] = fromId;
+		    w_lock[0] = 0;
+		    for(int id = 0; id < ThreadNum; id ++){
 			threadStatus[id] = false;
 			final int tid = id;
 			new Thread(new Runnable() {
@@ -129,7 +128,7 @@
 				}
 			}).start();
 		}
-		while(true){
+		    while(true){
 			boolean found = false;
 			try {
 				Thread.sleep(100);
@@ -144,10 +143,11 @@
 			}
 			if(!found)break;
 		}
-		dos_dat.flush();
-		dos_dat.close();
-		dos_idx.flush();
-		dos_idx.close();
+		    dos_dat.flush();
+		    dos_idx.flush();
+		}
+
+		
 		System.out.println("Build finished!");
 	}
 	
