--- /tmp/orig-7984080511804208532.java	2025-09-24 21:51:15.895305752 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url416ee5b4ee_martinmarinov_HonProj_tgz-pJ8-martin_gui_MainJ8/src/martin/quantum/McalcDescription.java	2025-09-24 21:51:13.931168450 +0000
@@ -86,11 +86,11 @@
 			p.setProperty("prop_in", inputs);
 			p.setProperty("prop_meas", measurements);
 			p.setProperty("prop_vars", variables);
+			try (final FileOutputStream out = new FileOutputStream(f)) {
+			    p.store(out, "---MBQC Pattern description---");
+			}
+
 			
-			
-			final FileOutputStream out = new FileOutputStream(f);
-			p.store(out, "---MBQC Pattern description---");
-			out.close();
 		
 	}
 	
