--- /tmp/orig-14998607466672304633.java	2025-09-24 22:48:17.818514352 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlc98c3b97d2_Trimax_venta_tgz-pJ8-com_gesoftware_venta_structures_map_MapTestJ8/src/com/gesoftware/venta/network/SafeNetworkTest.java	2025-09-24 22:48:15.626361102 +0000
@@ -50,7 +50,9 @@
     }
 
     private static void startClient() {
-        final SafeConnection connection = new SafeConnection("localhost", c_Port, c_Bits, new IClientHandler() {
+    com.gesoftware.venta.network.SafeConnection connection = null;
+    try {
+        connection = new SafeConnection("localhost", c_Port, c_Bits, new IClientHandler() {
             @Override
             public void onReceive(final Message message) {
                 LoggingUtility.info("Server answer: " + message.toString());
@@ -61,9 +63,19 @@
                 LoggingUtility.info("Connection lost: " + message);
             }
         });
-
         connection.connect();
         (new java.util.Timer("Client")).schedule(new Task(connection), 0, 1000);
+    } finally {
+        if (connection != null) {
+            try {
+                connection.disconnect();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
     }
 
     public static void main(final String args[]) {
