--- /tmp/orig-2446560984692313970.java	2025-09-24 22:17:22.792838000 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/Utils/ReadSaveData.java	2025-09-24 22:17:19.756625758 +0000
@@ -27,8 +27,7 @@
      */
     public static ArrayList<KitConfig> readKitConfig() {
     	ArrayList<KitConfig> kitConfigs = null;
-    	try{
-    		InputStream is = new FileInputStream("KitConfigBackup.sav");
+    	try(InputStream is = new FileInputStream("KitConfigBackup.sav")) {
     		InputStream buffer = new BufferedInputStream(is);
     		ObjectInput oi = new ObjectInputStream(buffer);
     		try {
