--- /tmp/orig-3287236939431681615.java	2025-09-24 22:01:43.323165669 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url4c41254592_topskychen_AuthTopk_tgz-pJ8-quickhull3d_QuickHull3DTestJ8/src/utility/DataFormater.java	2025-09-24 22:01:40.482967127 +0000
@@ -16,10 +16,14 @@
 	public static String output_file_name = "source/NE.dat";
 	
 	public static void genCa(){
-		try{
+	java.io.FileWriter __arodnap_temp1 = null;
+	java.io.FileReader __arodnap_temp0 = null;
+	try{
 			String line = null;
-			LineNumberReader lr = new LineNumberReader(new FileReader(input_file_name));
-			PrintWriter pw = new PrintWriter(new FileWriter(output_file_name));
+			__arodnap_temp0 = new FileReader(input_file_name);
+			LineNumberReader lr = new LineNumberReader(__arodnap_temp0);
+			__arodnap_temp1 = new FileWriter(output_file_name);
+			PrintWriter pw = new PrintWriter(__arodnap_temp1);
 			HashSet<String> cnt = new HashSet<String>();
 			try {
 				int line_num = 0;
@@ -59,6 +63,21 @@
 		} catch (IOException e) {
 			// TODO Auto-generated catch block
 			e.printStackTrace();
+		} finally {
+		    if (__arodnap_temp1 != null) {
+		        try {
+		            __arodnap_temp1.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
+		    if (__arodnap_temp0 != null) {
+		        try {
+		            __arodnap_temp0.close();
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
 		}
 		System.out.println("data format fin!");
 	}
