--- /tmp/orig-13157279619676200451.java	2025-09-24 22:08:51.545100022 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url70ed135da7_ralfbiedert_jcores_tgz-pJ8-sandbox_SimpleFilterTestJ8/src/net/jcores/script/util/console/JCoresConsole.java	2025-09-24 22:08:48.092858699 +0000
@@ -96,7 +96,9 @@
 	 * -systemout-and-systemerr-to.html
 	 */
 	private void redirectSystemStreams() {
-		OutputStream out = new OutputStream() {
+	java.io.OutputStream out = null;
+	try {
+	    out = new OutputStream() {
 			@Override
 			public void write(int b) throws IOException {
 				updateTextArea(String.valueOf((char) b));
@@ -112,8 +114,18 @@
 				write(b, 0, b.length);
 			}
 		};
+	    System.setOut(new PrintStream(out, true));
+	    System.setErr(new PrintStream(out, true));
+	} finally {
+	    if (out != null) {
+	        try {
+	            out.close();
+	        } catch (Exception e) {
+	            e.printStackTrace();
+	        }
+	    }
+	}
+
 
-		System.setOut(new PrintStream(out, true));
-		System.setErr(new PrintStream(out, true));
 	}
 }
