--- /tmp/orig-7671031724595305296.java	2025-09-24 21:54:06.407225271 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url47af6b1535_rolandotr_db_comparison_tgz-pJ8-methodology_HistoryJ8/src/utils/Report.java	2025-09-24 21:54:04.643101952 +0000
@@ -236,11 +236,12 @@
 
 	private static List<DBProtocol> getPoulidorProtocol(File f) throws NumberFormatException, IOException {
 		List<DBProtocol> protocols = new LinkedList<>();
-		BufferedReader reader = new BufferedReader(new FileReader(f));
-		String regex = "Poulidor-\\{(\\d+)\\}";
-		Pattern pattern = Pattern.compile(regex);
-		String retrievedText = null;
-		for (String line = reader.readLine(); line  != null; line = reader.readLine()){
+		try (java.io.FileReader __arodnap_temp0 = new FileReader(f)) {
+		    BufferedReader reader = new BufferedReader(__arodnap_temp0);
+		    String regex = "Poulidor-\\{(\\d+)\\}";
+		    Pattern pattern = Pattern.compile(regex);
+		    String retrievedText = null;
+		    for (String line = reader.readLine(); line  != null; line = reader.readLine()){
 			Matcher m = pattern.matcher(line);
 			while (m.find()) {
 				retrievedText = m.group(1);
@@ -249,7 +250,7 @@
 				protocols.add(p);
 			}
 		}
-		reader.close();
+		}
 		return protocols;
 	}
 
