--- /tmp/orig-12777754380648666405.java	2025-09-24 22:44:35.042940434 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlb5d5be4365_abmargb_oursim_ext_tgz-pJ8-br_edu_ufcg_lsd_oursim_experiments_RemoteNoFScenarioGeneratorJ8/src/br/edu/ufcg/lsd/oursim/queue/FileEventProxyTest.java	2025-09-24 22:44:32.942793615 +0000
@@ -61,8 +61,20 @@
 	
 	@Test(expected=IllegalArgumentException.class)
 	public void testBadFormat() {
-		new FileEventProxy(
-				getClass().getResourceAsStream("event-bad.conf"));
+	java.io.InputStream __arodnap_temp0 = null;
+	try {
+    __arodnap_temp0 = getClass().getResourceAsStream("event-bad.conf");
+    new FileEventProxy(
+				__arodnap_temp0);
+} finally {
+    if (__arodnap_temp0 != null) {
+        try {
+            __arodnap_temp0.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+}
 	}
 	
 }
