--- /tmp/orig-6992054599162370062.java	2025-09-24 21:35:40.249915928 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url0d38a10ee7_diegopliebana_GEBT_tgz-pJ8-ch_idsia_scenarios_test_EvolveWithChangingSeedsJ8/src/ch/idsia/utils/StatisticalSummary.java	2025-09-24 21:35:37.153699577 +0000
@@ -251,10 +251,10 @@
     }
 
     public static StatisticalSummary load(String path) {
-        try {
+        try (java.io.FileInputStream __arodnap_temp0 = new FileInputStream(path)) {
             ObjectInputStream ois =
                     new ObjectInputStream(
-                            new FileInputStream(path));
+                            __arodnap_temp0);
             StatisticalSummary ss = (StatisticalSummary) ois.readObject();
             ois.close();
             return ss;
