--- /tmp/orig-1548404433999531297.java	2025-09-24 22:27:13.378122225 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url80e39b1957_tobikster_IJN_headers_recognition_tgz-pJ8-org_maltparser_examples_CreatePhraseStructureGraphJ8/src/org/maltparser/ml/lib/LibSvm.java	2025-09-24 22:27:09.873877263 +0000
@@ -29,8 +29,8 @@
 	public LibSvm(InstanceModel owner, Integer learnerMode) throws MaltChainedException {
 		super(owner, learnerMode, "libsvm");
 		if (learnerMode == CLASSIFY) {
-			try {
-			    ObjectInputStream input = new ObjectInputStream(getInputStreamFromConfigFileEntry(".moo"));
+			try (java.io.InputStream __arodnap_temp0 = getInputStreamFromConfigFileEntry(".moo")) {
+			    ObjectInputStream input = new ObjectInputStream(__arodnap_temp0);
 			    try {
 			    	model = (MaltLibModel)input.readObject();
 			    } finally {
