--- /tmp/orig-3258232856547953005.java	2025-09-24 21:54:19.496140243 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url47af6b1535_rolandotr_db_comparison_tgz-pJ8-methodology_HistoryJ8/src/utils/Report.java	2025-09-24 21:54:17.764019161 +0000
@@ -294,11 +294,12 @@
 
 	private static List<DBProtocol> getKimAndAvoineProtocol(File f) throws NumberFormatException, IOException {
 		List<DBProtocol> protocols = new LinkedList<>();
-		BufferedReader reader = new BufferedReader(new FileReader(f));
-		String regex = "KA-\\{(\\d+), (\\d\\.\\d+)\\}";
-		Pattern pattern = Pattern.compile(regex);
-		String retrievedText = null;
-		for (String line = reader.readLine(); line  != null; line = reader.readLine()){
+		try (java.io.FileReader __arodnap_temp0 = new FileReader(f)) {
+		    BufferedReader reader = new BufferedReader(__arodnap_temp0);
+		    String regex = "KA-\\{(\\d+), (\\d\\.\\d+)\\}";
+		    Pattern pattern = Pattern.compile(regex);
+		    String retrievedText = null;
+		    for (String line = reader.readLine(); line  != null; line = reader.readLine()){
 			Matcher m = pattern.matcher(line);
 			while (m.find()) {
 				KimAndAvoineProtocol p = new KimAndAvoineProtocol(Double.parseDouble(m.group(2)));
@@ -307,7 +308,7 @@
 				protocols.add(p);
 			}
 		}
-		reader.close();
+		}
 		return protocols;
 	}
 
