--- /tmp/orig-1108894406088834278.java	2025-09-24 21:44:48.760243651 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/fri/patterns/interpreter/parsergenerator/parsertables/LRParserTables.java	2025-09-24 21:44:46.168062451 +0000
@@ -55,13 +55,22 @@
 			{ "L", "'id'" },
 			{ "R", "L", },
 		};
+		fri.patterns.interpreter.parsergenerator.parsertables.LRParserTables p = null;
 
 		try	{
-			LRParserTables p = new LRParserTables(new Syntax(syntax));
+			p = new LRParserTables(new Syntax(syntax));
 			p.dump(System.err);
 		}
 		catch (Exception e)	{
 			e.printStackTrace();
+		} finally {
+		    if (p != null) {
+		        try {
+		            p.toSourceFile(null);
+		        } catch (Exception e) {
+		            e.printStackTrace();
+		        }
+		    }
 		}
 	}
 	
