--- /tmp/orig-13396218641076729535.java	2025-09-24 22:45:03.352919543 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/utils/Configuration.java	2025-09-24 22:45:01.360780275 +0000
@@ -38,13 +38,23 @@
         {
             type = "";
         }
+        java.io.FileInputStream __arodnap_temp0 = null;
         try
         {
-            return load(new FileInputStream(file), type);
+        __arodnap_temp0 = new FileInputStream(file);
+        return load(__arodnap_temp0, type);
         }
         catch (final FileNotFoundException e)
         {
             e.printStackTrace();
+        } finally {
+            if (__arodnap_temp0 != null) {
+                try {
+                    __arodnap_temp0.close();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
         }
         return false;
     }
