--- /tmp/orig-355905655410401146.java	2025-09-24 21:38:11.800505629 +0000
+++ /home/smala009/RLF/cf-rlc/june2020_dataset_NJR/url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpRequestUri.java	2025-09-24 21:38:09.700358880 +0000
@@ -110,16 +110,28 @@
 	}
 
 	public void testEncode002() throws IOException {
-		HttpGetRequestUri uri = new HttpGetRequestUri("/announce");
-		uri.putVale("01", "value01");
+	net.hetimatan.util.http.HttpGetRequestUri uri = null;
+	try {
+	    uri = new HttpGetRequestUri("/announce");
+	    uri.putVale("01", "value01");
+	    CashKyoroFile output = new CashKyoroFile(512);
+	    uri.encode(output.getLastOutput());
+	    output.seek(0);
+	    byte[] buffer = new byte[(int)output.length()];
+	    int len = output.read(buffer);
+	    String tag = new String(buffer, 0, len);
+	    assertEquals("/announce?01=value01", tag);
+	} finally {
+	    if (uri != null) {
+	        try {
+	            uri.createEncode(null);
+	        } catch (Exception e) {
+	            e.printStackTrace();
+	        }
+	    }
+	}
+
 
-		CashKyoroFile output = new CashKyoroFile(512);
-		uri.encode(output.getLastOutput());
-		output.seek(0);
-		byte[] buffer = new byte[(int)output.length()];
-		int len = output.read(buffer);
-		String tag = new String(buffer, 0, len);
-		assertEquals("/announce?01=value01", tag);
 	}
 
 	public void testEncode003() throws IOException {
